evm_loader/program/src/account/sysvar.rs

```diff
@@ -0,0 +1,143 @@
+        while (compute_unit_limit.is_none() || compute_unit_price.is_none()) && idx < max_idx {
+            let cur_ixn = instructions::load_instruction_at_checked(idx, self.0)?;
+
+            // Skip all instructions that do not target Compute Budget Program.
+            if cur_ixn.program_id != compute_budget_account_pubkey {
+                idx += 1;
+                continue;
+            }
+
+            // As of now, data of ComputeBudgetInstruction is always non-empty.
+            // This is a sanity check to have a safe future-proof implementation.
+            let tag = cur_ixn.data.first().unwrap_or(&0);
+            match *tag {
+                COMPUTE_UNIT_LIMIT_TAG => {
+                    compute_unit_limit = Some(u32::from_le_bytes(
+                        cur_ixn.data[1..].try_into().map_err(|_| {
+                            Error::PriorityFeeParsingError(
+                                "Invalid format of compute unit limit.".to_string(),
+                            )
+                        })?,
+                    ));
+                }
+                COMPUTE_UNIT_PRICE_TAG => {
+                    compute_unit_price = Some(u64::from_le_bytes(
+                        cur_ixn.data[1..].try_into().map_err(|_| {
+                            Error::PriorityFeeParsingError(
+                                "Invalid format of compute unit price.".to_string(),
+                            )
+                        })?,
+                    ));
+                }
+                _ => (),
+            }
+            idx += 1;
+        }
+
+        if compute_unit_price.is_none() {
+            return Err(Error::PriorityFeeNotSpecified);
+        }
+
+        // Caller may not specify the compute unit limit, the default should take effect.
+        if compute_unit_limit.is_none() {
+            compute_unit_limit = Some(DEFAULT_COMPUTE_UNIT_LIMIT);
+        }
+
+        // Both are not none, it's safe to unwrap.
+        Ok((compute_unit_limit.unwrap(), compute_unit_price.unwrap()))
+    }
+
+    /// Checks that priority fee as set by the Operator is accurate to what User set as `max_priority_fee_per_gas`.
+    pub fn validate_priority_fee(
+        &self,
+        trx_max_priority_fee_per_gas: U256,
+        trx_max_fee_per_gas: U256,
+    ) -> Result<(), Error> {
+        let (cu_limit, cu_price) = self.get_compute_budget_priority_fee()?;
+        let priority_fee: U256 = cu_price
+            .checked_mul(u64::from(cu_limit))
+            .ok_or(Error::PriorityFeeError(
+                "cu_limit * cu_price overflow".to_string(),
+            ))?
+            .into();
+        let base_fee_per_gas = trx_max_fee_per_gas - trx_max_priority_fee_per_gas;
+
+        let actual_priority_fee_per_gas = base_fee_per_gas
+            .checked_mul(priority_fee)
+            .and_then(|res| res.checked_div(CONVERSION_DIVISOR.into()))
+            .ok_or(Error::PriorityFeeError("base_fee_per_gas * ".into()))?;
```

Comment:
I think this string may be missing an operator

Reply:
done